Adding element type page in from wiki

adamdewey opened the pull request:
Re-ordered a few menu items too to try and align more with https://diataxis.fr/

CarlosNZ reviewed (changes requested):
Cool, thanks Adam, will have a proper look soon. Would you also mind updating the page in the original page in the dev docs to just have a note linking to this page? (so we know to only maintain one page going forward)

Should be able to do that in this PR too.

adamdewey commented:
> Cool, thanks Adam, will have a proper look soon. Would you also mind updating the page in the original page in the dev docs to just have a note linking to this page? (so we know to only maintain one page going forward)
> 
> Should be able to do that in this PR too.

Sure - not quite sure how I update a PR though?

CarlosNZ commented:
> Sure - not quite sure how I update a PR though?

Just make another commit on this branch and push it, that's all :)

adamdewey commented:
> > Sure - not quite sure how I update a PR though?
> 
> Just make another commit on this branch and push it, that's all :)

Thanks Carl - for some reason GitHub Desktop kept telling me I was making dirty subproject changes, so I just updated the wiki using the UI!

CarlosNZ commented:
>  so I just updated the wiki using the UI!

That won't work -- as soon as someone makes a change to the internal docs, the "wiki" page will get over-written next update. (Docs -> wiki changes should only go in one direction)

You need to update `Element-Type-Specs.md` in /docs/internal. Don't worry about making it update the wiki just yet. Our normal workflow is to update the docs, then when it's approved/merged update the wiki (which we do using the `yarn push_docs` command).

If you're getting dirty subproject errors, that just means your submodule (the _wiki folder) doesn't match the commit that that the "outer" repo has on record for it. You can ignore it, but it you want it to be correct, just go into the "_wiki" folder at the command line and use `git pull`. (If you want further clarification around this, just give me a call -- it took me ages to understand what's going on with the submodule commits).

adamdewey commented:
> > so I just updated the wiki using the UI!
> 
> That won't work -- as soon as someone makes a change to the internal docs, the "wiki" page will get over-written next update. (Docs -> wiki changes should only go in one direction)
> 
> You need to update `Element-Type-Specs.md` in /docs/internal. Don't worry about making it update the wiki just yet. Our normal workflow is to update the docs, then when it's approved/merged update the wiki (which we do using the `yarn push_docs` command).
> 
> If you're getting dirty subproject errors, that just means your submodule (the _wiki folder) doesn't match the commit that that the "outer" repo has on record for it. You can ignore it, but it you want it to be correct, just go into the "_wiki" folder at the command line and use `git pull`. (If you want further clarification around this, just give me a call -- it took me ages to understand what's going on with the submodule commits).

Ah ok, gotcha - thanks for that, I was updating in /docs/_wiki which was why it was giving me the dirty. 

Have now updated /doc/internal 

Cheers

CarlosNZ commented on docs/internal/Element-Type-Specs.md:
```diff
@@ -1,3 +1,5 @@
+_**Note: this document has been superseded by the zola site https://docs.mflow.nz/**_
```
```suggestion
_**Note: Element type reference is available in the [public docs](https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>) **_
```

CarlosNZ commented on docs/internal/Element-Type-Specs.md:
```diff
@@ -1,3 +1,5 @@
+_**Note: this document has been superseded by the zola site https://docs.mflow.nz/**_
+
 _Ongoing authoritative reference of Template Question/Element types, including input parameters and response type (shape). Please ensure this document matches the current implementation at all times._
```
Delete everything else -- no point have two slightly different versions hanging around.

adamdewey replied:
Cool, just deleted and committed

CarlosNZ commented on docs/public/content/reference/forms/index.md:
````diff
@@ -0,0 +1,661 @@
+
+_For specifying where the list of questions is broken into UI pages/steps. The **previous** question of this element will be the **last** element on a page_
+
+#### Input parameters
+
+- ~~**pageBreakValidityCheck**: `boolean` -- If `true`, the user cannot proceed to the next page unless _all_ questions on the current page have passed validation~~
+  - ~~default: `false`~~
+
+---
+
+```
+
+```
````
This can go.

adamdewey replied:
This has gone also now

CarlosNZ reviewed (changes requested):
Great, just a couple of tiny things.

CarlosNZ commented on docs/internal/Element-Type-Specs.md:
```diff
@@ -1,626 +1,3 @@
-_Ongoing authoritative reference of Template Question/Element types, including input parameters and response type (shape). Please ensure this document matches the current implementation at all times._
+_**Note: Element type reference is available in the [public docs](https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>) **_
```
`https://docs.mflow.nz/<put-the-exact-url-here-if-you-know-it>`

Do you know it? Well -- do ya?

adamdewey replied:
Oh! Haha, should be there now

CarlosNZ reviewed (approved):
Nice :)
